dartagnan/src/main/java/com/dat3m/dartagnan/wmm/axiom/Empty.java

```diff
@@ -20,22 +21,25 @@ public Empty(Relation rel) {
     }
 
     @Override
-    public TupleSet getEncodeTupleSet(){
-        return rel.getMaxTupleSet();
+    protected Set<Tuple> getEncodeTupleSet(Context analysisContext) {
+        final RelationAnalysis ra = analysisContext.get(RelationAnalysis.class);
+        return ra.getKnowledge(rel).getMaySet();
     }
 
     @Override
-    public BooleanFormula consistent(Set<Tuple> toBeEncoded, EncodingContext ctx) {
+    public BooleanFormula consistent(EncodingContext ctx) {
     	BooleanFormulaManager bmgr = ctx.getBooleanFormulaManager();
 		BooleanFormula enc = bmgr.makeTrue();
-        for (Tuple tuple : toBeEncoded) {
-            enc = bmgr.and(enc, bmgr.not(ctx.edge(rel, tuple)));
+        final RelationAnalysis ra = ctx.getAnalysisContext().get(RelationAnalysis.class);
+        final EncodingContext.EdgeEncoder edge = ctx.edge(rel);
+        for (Tuple tuple : ra.getKnowledge(rel).getMaySet()) {
```

Comment:
I think `toBeEncoded` used to be the encodeTupleSet. Here we iterate over the may set. Is this correct?

Reply:
I think this is fine here, because `encodeSet=mayset` for the `Emptiness` axiom. It might be nice to have a comment that mentions this.